### PR TITLE
Add a MARGINAL_FEE constant for the ZIP 317 marginal fee

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK modulo comment, but please add a release notes entry, since this is a public API addition. Also please change the commit message to match the PR title (and squash all commits), since there are no uses in tests.

(I'm not sure we want clients to be calculating ZIP 317 fees themselves rather than using the `FeeRule`, but it seems relatively harmless to add the constant, and we already have a constant for the minimum fee.)

File: zcash_primitives/src/transaction/fees/zip317.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use crate::{
 ///
 /// [ZIP 317]: https//zips.z.cash/zip-0317
 pub const MINIMUM_FEE: Amount = Amount::const_from_i64(10_000);
+pub const MARGINAL_FEE: Amount = Amount::const_from_i64(5_000);
```
Comment (daira) [inline zcash_primitives/src/transaction/fees/zip317.rs:22]: Remember that each public item must be separately documented.

```suggestion

/// The marginal fee using the standard [ZIP 317] constants. Equivalent to
/// `FeeRule::standard().marginal_fee()`, but as a constant.
///
/// [ZIP 317]: https//zips.z.cash/zip-0317
pub const MARGINAL_FEE: Amount = Amount::const_from_i64(5_000);
```

Comment (nuttycom) [reply zcash_primitives/src/transaction/fees/zip317.rs:22]: Oh hey, I wanted to make this change just yesterday!

File: zcash_primitives/src/transaction/fees/zip317.rs [hunk 2]
```diff
@@ -20,6 +20,12 @@ use crate::{
 /// [ZIP 317]: https//zips.z.cash/zip-0317
 pub const MINIMUM_FEE: Amount = Amount::const_from_i64(10_000);
 
+/// The marginal fee using the standard [ZIP 317] constants. Equivalent to
+/// `FeeRule::standard().marginal_fee()`, but as a constant.
```
Comment (nuttycom) [inline zcash_primitives/src/transaction/fees/zip317.rs:24]:
```suggestion
/// The marginal fee using the standard [ZIP 317] constants. 
```

Comment (nuttycom) [reply zcash_primitives/src/transaction/fees/zip317.rs:24]: `FeeRule::standard` could change at some point and this would then be incorrect.

**Review by nuttycom — APPROVED** [review]
utACK 0e3b0e6d213c0eb02b8ef085f95fe2daddfc271d

**Review by nuttycom — APPROVED** [review]
re-utACK

**Pull request merged**